### PR TITLE
mfreitassm/c01-git03

### PR DESCRIPTION
**Pull request by mfreitassm**
# Description

Exercise (e.g. C01-E01): 

Student Name: Mariana Freitas 

> Answer exercise c01-git03.

## Type of change

- [x] Exercise Submission (one exercise per PR)
  - [x] This PR is just for **one** exercise of a class
  - [x] PR name follows the pattern `<github-username>/<exercise-number>`
  - [x] Added exercise files inside folder `/classes/<class- number>/exercises/<exercise-number>/<github-username>/`
	- [ ] Review [exercise submission steps](./README.md#Exercises)
- [ ] Class content (instructors/admins)
- [ ] Documentation update
- [ ] Repo management

### CODE REVIEW

File: classes/01class/exercises/c01-git03/mfreitassm/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+Yes, before added the big file (100M) repo size was 168k and after added was 201M.
+
+2. What is the reason for this problem?
+
+A file copy was stored on the folder /.git/objects as history.
+
+3. How do you remove something from Git history after it is pushed to the remote repository? Which commands would you use? 
+
+Git history will need to be rewrited.
+git filter-branch --index-filter "git rm -rf --cached --ignore-unmatch path_to_file" HEAD
```
Comment (denstorti) [inline classes/01class/exercises/c01-git03/mfreitassm/README.md:16]: In a few words, what's it doing?

File: classes/01class/exercises/c01-git03/mfreitassm/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+2. What is the reason for this problem?
+
+A file copy was stored on the folder /.git/objects as history.
+
+3. How do you remove something from Git history after it is pushed to the remote repository? Which commands would you use? 
+
+Git history will need to be rewrited.
+git filter-branch --index-filter "git rm -rf --cached --ignore-unmatch path_to_file" HEAD 
+git push -all
```
Comment (denstorti) [inline classes/01class/exercises/c01-git03/mfreitassm/README.md:17]: if the commits were rewritten, here you need to `--force` the push

**Comment by denstorti** [comment]
Hello, any update on this one?

File: classes/01class/exercises/c01-git03/mfreitassm/README.md (hunk 1 above)
Comment (mfreitassm) [reply classes/01class/exercises/c01-git03/mfreitassm/README.md:16]: Hi Denis, I am sorry about the delay!, 
I had been off  for the last 2 weeks due to some personal issues .

To answer:
Git filter-branch will do is for each commit in the specified branch, apply filters to the snapshot, and create a new commit. The new commit’s parent will be the filtered version of the old commit’s parent. So it creates a parallel commit DAG.

File: classes/01class/exercises/c01-git03/mfreitassm/README.md (hunk 2 above)
Comment (mfreitassm) [reply classes/01class/exercises/c01-git03/mfreitassm/README.md:17]: Okay!
Thanks!

File: classes/01class/exercises/c01-git03/mfreitassm/README.md (hunk 1 above)
Comment (caiocezart) [reply classes/01class/exercises/c01-git03/mfreitassm/README.md:16]: Can you update your answers with that?

**Comment by roperto** [comment]
Hi @mfreitassm -- let me know on slack if you are having problems submitting the exercises, otherwise your request might end up being automatically deleted. Cheers

**Pull request merged**